lib/core/src/recover/model.rs

```diff
@@ -168,25 +196,41 @@ pub(crate) struct RecoveredOnchainDataChainReceive {
     /// BTC tx initiated by the SDK to a user-chosen address, in case the initial funds have to be refunded.
     pub(crate) btc_refund_tx_id: Option<HistoryTxId>,
 }
+// TODO: We have to be careful around overwriting the Refundable or RefundPending state, as this swap monitored
+// after the expiration of the swap and if new funds are detected on the lockup script they are either refunded
```

Comment:
Maybe not a problem: https://github.com/breez/breez-sdk-liquid/blob/e4e1ed95c1775b32676c83afc5d5ec43bef2ac5a/lib/core/src/recover/recoverer.rs#L588-L633

We only track the first two txs (lockup + claim or refund), and even ensure there's no overlap between claim and refund tx, so we should not be able to get to the condition where we mark it as Refundable/RefundPending by mistake (e.g. if a new tx is detected at that address)

Reply:
I think the recoverer should be able to handle this, and make sure the `btc_refund_tx_id` is aways the latest one